.github/workflows/upload-release-builds.yml

```diff
@@ -0,0 +1,124 @@
+          shell: cmd
+          build-type: Release
+          run-on-prs: false
+        - os: ubuntu-20.04
+          shell: bash
+          build-type: Debug
+          run-on-prs: false
+        - os: ubuntu-20.04
+          shell: bash
+          build-type: RelWithDebInfo
+          run-on-prs: true
+        - os: ubuntu-20.04
+          shell: bash
+          build-type: Release
+          run-on-prs: false
+          # MacOS toolchain doesn't search /usr/local by default:
+          # https://gitlab.kitware.com/cmake/cmake/-/issues/19120
+          # The CMake FindPython module's Python::Python target (used
+          # via pybind11::embed in python-bridge-test) transitively adds
+          # linker flags to system libs, which fail due to this issue.
+        - os: macos-11
+          preamble: export LDFLAGS="-L/usr/local/lib"
+          shell: bash
+          build-type: Debug
+          run-on-prs: false
+        - os: macos-11
+          preamble: export LDFLAGS="-L/usr/local/lib"
+          shell: bash
+          build-type: RelWithDebInfo
+          run-on-prs: true
+        - os: macos-11
+          preamble: export LDFLAGS="-L/usr/local/lib"
+          shell: bash
+          build-type: Release
+          run-on-prs: false
+    defaults:
+      run:
+        # Annoyingly required here since `matrix` isn't available in
+        # the `shell` property of individual steps.
+        shell: ${{ matrix.config.shell }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Bootstrap
+        uses: ./.github/bootstrap_platform
+
+      # We don't want to publish the test build, it gets too big.
+      - name: Build and Install (No tests)
+        run: >
+          ${{ matrix.config.preamble }}
+
+          cmake -S . -B build -G Ninja
+          --install-prefix ${{ github.workspace }}/dist/${{ matrix.config.os }}${{ matrix.config.build-type }}
+          --toolchain ${{ github.workspace }}/.conan/conan_paths.cmake
+          -DCMAKE_BUILD_TYPE=${{matrix.config.build-type}}
+
+          cmake --build build
+
+          cmake --install build
+
+      - name: Get compiler information
+        id: compiler_info
+        shell: bash
+        # Search the cmake info for the compiler information, remove the
+        # env var labels, and dequote it.
+        run: |
+          cd build
+          compiler_version=$(cmake --system-information | grep -E "^CMAKE_CXX_COMPILER_VERSION " | cut -d ' ' -f2 | tr -d '"')
+          compiler_id=$(cmake --system-information | grep -E "^CMAKE_CXX_COMPILER_ID " | cut -d ' ' -f2 | tr -d '"')
+          python_version=$(python -V | sed 's/ /-/')
+          echo "compiler_version=$compiler_version" >> $GITHUB_OUTPUT
+          echo "compiler_id=$compiler_id" >> $GITHUB_OUTPUT
+          echo "python_version=$python_version" >> $GITHUB_OUTPUT
+
+      - name: Bundle and Upload Artifact
+        #Use inbuilt powershell commands because `zip` isn't on windows.
+        shell: pwsh
```

Comment:
I did consider it, but eventually settled on keeping them split to avoid having a massive chunk of shell script that did two things, especially with how failure reporting works in actions. Not to mention having to learn how to do the filtering of the variables in powershell, which I don't know how to do.

Reply:
OK no worries, good as-is.